jax/numpy/lax_numpy.py

```diff
@@ -1160,8 +1160,11 @@ def ravel(a, order="C"):
 
 @_wraps(onp.unravel_index, lax_description=_UNRAVEL_INDEX_DOC)
 def unravel_index(flat_index, shape):
+  flat_index = asarray(flat_index)
   sizes = pad(shape, (0, 1), constant_values=1)
   cumulative_sizes = cumprod(sizes[::-1])[::-1]
+  # Add enough trailing dims to avoid conflict with flat_index
+  cumulative_sizes = cumulative_sizes.reshape([-1] + [1] * flat_index.ndim)
```

Comment:
@jakevdp if you know a cleaner way to add trailing dim(s)...

Reply:
A useful pattern here would be to store the shape of the input, then `ravel()` it so the computation only has to handle 1D, then reshape the outputs before returning.

Reply:
Something like this?

```
idx = clipped_index.ravel() % cumulative_sizes[:-1] // cumulative_sizes[1:]
return tuple(idx.reshape([-1] + list(flat_index.shape)))
```

To my eye, the current implementation is slightly cleaner. Unless you are seeing other benefits, inclined to leave as is.

Reply:
No, I was saying that you ravel ``flat_index`` above so you can avoid the complicated reshape expression here, and then you return ``tuple([d.reshape(flat_index.shape) for d in idx])``

Reply:
Sure could do multiple reshapes to avoid `[-1] + list(flat_index.shape)` . The current implementation still feels a bit cleaner to me.